src/libraries/System.Data.Common/src/System/Data/Common/DbConnection.cs

```diff
@@ -143,21 +143,166 @@ public virtual void EnlistTransaction(System.Transactions.Transaction? transacti
+        /// <exception cref="ArgumentException">
+        /// <paramref name="collectionName" /> is specified as <see langword="null" />.
+        /// </exception>
+        /// <remarks>
+        /// If the connection is associated with a transaction, executing <see cref="GetSchema(string)" /> calls may cause
+        /// some providers to throw an exception.
+        /// </remarks>
         public virtual DataTable GetSchema(string collectionName)
         {
             throw ADP.NotSupported();
         }
 
+        /// <summary>
+        /// Returns schema information for the data source of this <see cref="DbConnection" /> using the specified
+        /// string for the schema name and the specified string array for the restriction values.
+        /// </summary>
+        /// <param name="collectionName">Specifies the name of the schema to return.</param>
+        /// <param name="restrictionValues">Specifies a set of restriction values for the requested schema.</param>
+        /// <returns>A <see cref="DataTable" /> that contains schema information.</returns>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="collectionName" /> is specified as <see langword="null" />.
+        /// </exception>
+        /// <remarks>
+        /// <para>
+        /// The <paramref name="restrictionValues" /> parameter can supply n depth of values, which are specified by the
+        /// restrictions collection for a specific collection. In order to set values on a given restriction, and not
+        /// set the values of other restrictions, you need to set the preceding restrictions to null and then put the
+        /// appropriate value in for the restriction that you would like to specify a value for.
+        /// </para>
+        /// <para>
+        /// An example of this is the "Tables" collection. If the "Tables" collection has three restrictions (database,
+        /// owner, and table name) and you want to get back only the tables associated with the owner "Carl", you must
+        /// pass in the following values at least: null, "Carl". If a restriction value is not passed in, the default
+        /// values are used for that restriction. This is the same mapping as passing in null, which is different from
+        /// passing in an empty string for the parameter value. In that case, the empty string ("") is considered to be
+        /// the value for the specified parameter.
+        /// </para>
+        /// <para>
+        /// If the connection is associated with a transaction, executing <see cref="GetSchema(string, string[])" />
+        /// calls may cause some providers to throw an exception.
+        /// </para>
+        /// </remarks>
         public virtual DataTable GetSchema(string collectionName, string?[] restrictionValues)
         {
             throw ADP.NotSupported();
         }
 
+        /// <summary>
+        /// This is the asynchronous version of <see cref="GetSchema()" />.
```

Comment:
All of this should be in remarks I think.  The summary should be similar to the sync version.

Reply:
(Same for the others)

Reply:
@stephentoub you're right, the docs for all async methods currently point to the sync method and add some notes on the default implementation - not ideal. I've opened #39154 to clean this up globally in System.Data, will take care of that after this PR.